State.cpp

```diff
@@ -0,0 +1,248 @@
+#include "BreathData.h"
+#include "State.h"
+#include "Constants.h"
+#include "Display.h"
+#include "Flow.h"
+#include "Pressure.h"
+#include "Valve.h"
+#include "ProportionalValve.h"
+#include "AlarmManager.h"
+#include "State.h"
+
+// Initialize singleton instances
+OffState* OffState::instance = 0;
+InsStateVC* InsStateVC::instance = 0;
+InsHoldStateVC* InsHoldStateVC::instance = 0;
+ExpStateVC* ExpStateVC::instance = 0;
+PeepStateVC* PeepStateVC::instance = 0;
+ExpHoldStateVC* ExpHoldStateVC::instance = 0;
+
+BreathData* State::breath;
+
+/*********************************************************/
+/* OffState                                              */
+/*********************************************************/
+
+// performs actions necessary upon entering a state
+static State* OffState::begin(BreathData b) {
+    if (instance==0) instance = new OffState();  // one-time setup
+    breath = &b;                                 // store reference to breath data
```

Comment:
This won't work. You are storing the address of a local variable, which will go out of scope immediately on return.  The `BreathData` argument needs to be a pointer.  But really, it need not exist, since the base class already has a `breath` object, there is no need to set it in every state.

Reply:
It's been a long while since I've programmed in C++, so my Java-based thinking is getting me in trouble.  I just want to initialize the base-class breath object for the first time.  Not sure where to put that.